Objective-C/CBLReplicator.mm

```diff
@@ -165,17 +167,38 @@ - (void) start {
     }
 }
 
-- (void) retry: (BOOL)reset {
+- (void) resetAndScheduleRetry {
+    [self resetRetryCount];
+    [self scheduleRetry: 0];
+}
+
+- (void) scheduleRetry: (NSTimeInterval)delayInSeconds {
+    [CBLTimer cancel: _retryTimer];
+    
+    __weak CBLReplicator *weakSelf = self;
+    _retryTimer = [CBLTimer scheduleIn: _dispatchQueue
+                   after: delayInSeconds
+                   block:^{
+        [weakSelf _retry];
+    }];
+}
+
+- (void) resetRetryCount {
+    CBLLogVerbose(Sync, @"%@: Resetting the retry count", self);
+    _retryCount = 0;
+}
+
+- (void) _retry {
     CBL_LOCK(self) {
-        CBLLogInfo(Sync, @"%@: Retrying...", self);
-        if (reset)
-            _retryCount = 0;
-        if (_repl || _state <= kCBLStateStopping) {
-            CBLLogInfo(Sync, @"%@: Ignore retrying (state = %d, status = %d)",
-                       self, _state, _rawStatus.level);
-            return;
+        if (_state == kCBLStateOffline || _state == kCBLStateSuspended) {
```

Comment:
Not sure if we need this if check is required. It seems like `if (_repl || _state <= kCBLStateStopping)` has already covered it.

Reply:
yes. you are right.